go/login/v2/login_test.go

```diff
@@ -0,0 +1,245 @@
+		fmt.Println("authorized")
+	} else {
+		fmt.Println("forbidden")
+	}
+
+	// Output:
+	// Message: myhost/user=root
+	// authorized
+}
+
+func ptr(i uint8) *uint8 {
+	return &i
+}
+
+type testVector struct {
+	index                  int
+	alice                  string
+	bob                    string
+	keyIndex               *uint8
+	messageTagPrefixLength uint8
+	minResponseLength      *uint8
+	msg                    *v2.Message
+
+	challenge string
+	response  string
+}
+
+var testVectors = []testVector{
+	{
+		index:                  1,
+		alice:                  "77076d0a7318a57d3c16c17251b26645df4c2f87ebc0992ab177fba51db92c2a",
+		bob:                    "5dab087e624a8a4b79e17f8b83800ee66f3bb1292618b6fd1c2f8b27ff88e0eb",
+		keyIndex:               ptr(0),
+		messageTagPrefixLength: 3,
+		minResponseLength:      ptr(12),
+		msg: &v2.Message{
+			HostIDType: "mytype",
+			HostID:     "myhost",
+			Action:     "root",
+		},
+		challenge: "v2/gIUg8AmJMKdUdIt93LQ-91oNvzoNJjga9OukqY6qm05qlyPH/mytype:myhost/root/",
+		response:  "BB4BYjXonlIRtXZORkQ5bF5xTZwW6o60ylqfCuyAHTQ=",
+	},
+	{
+		index: 2,
+		alice: "fee1deadfee1deadfee1deadfee1deadfee1deadfee1deadfee1deadfee1dead",
+		bob:   "b105f00db105f00db105f00db105f00db105f00db105f00db105f00db105f00d",
+		msg: &v2.Message{
+			HostID: "myhost",
+			Action: "exec=/bin/sh",
+		},
+		challenge: "v2/R4cvQ1u4uJ0OOtYqouURB07hleHDnvaogAFBi-ZW48N2/myhost/exec=%2Fbin%2Fsh/",
+		response:  "ZmxczN4x3g4goXu-A2AuuEEVftgS6xM-6gYj-dRrlis=",
+	},
+}
+
+func TestSpec(t *testing.T) {
+	for _, tc := range testVectors {
+		t.Run(fmt.Sprintf("vector-%d", tc.index), func(t *testing.T) {
+			alice := unhexPrivateKey(tc.alice)
+			bob := unhexPrivateKey(tc.bob)
+			bobPub, err := bob.Public()
+			if err != nil {
+				t.Fatalf("private key from test vector broken: %v", err)
+			}
+
+			c := &v2.Challenger{
+				PublicKey:              bobPub,
+				KeyIndex:               tc.keyIndex,
+				MessageTagPrefixLength: tc.messageTagPrefixLength,
+				RNG:                    bytes.NewBuffer(alice[:]),
+			}
+			if tc.minResponseLength != nil {
+				c.MinResponseLength = *tc.minResponseLength
+			}
+			cc, err := c.Challenge(tc.msg)
+			if err != nil {
+				t.Fatalf("Challenge generation failed: %v", err)
+			}
+			gotChallenge := cc.Encode()
+			if gotChallenge != tc.challenge {
+				t.Errorf("Unexpected encoding:\ngot:\t%s\nwant:\t%s", gotChallenge, tc.challenge)
+			}
+
+			r, err := v2.NewResponder(map[uint8]*glome.PrivateKey{0: bob})
+			if err != nil {
+				t.Fatalf("NewResponder() failed: %v", err)
+			}
+			sc, err := r.Accept(tc.challenge)
+			if err != nil {
+				t.Fatalf("Accept(%q) failed: %v", gotChallenge, err)
+			}
+			if !reflect.DeepEqual(tc.msg, sc.Message) {
+				t.Errorf("Responder parsed wrong message: got %#v, want %#v", sc.Message, tc.msg)
+			}
+
+			if sc.Response != tc.response {
+				t.Errorf("Responder generated wrong response: got %q, want %q", sc.Response, tc.response)
+			}
+
+			min := uint8(10)
```

Comment:
You don't need it to be uint8 for slicing, right? So should this just be an int and a cast when assigning it from `*tc.minResponseLength`?